main.tf

```diff
@@ -16,17 +16,24 @@ variable "role" {
   default     = "0"
 }
 
+variable "region" {
+  description = "Region to work in"
+  default     = "0"
+}
+
 variable "dependency_ids" {
   description = "IDs or ARNs of any resources that are a dependency of the resource created by this module."
   type        = "list"
   default     = []
 }
 
 data "aws_caller_identity" "id" {}
+data "aws_region" "current" {}
 
 locals {
   account_id      = "${var.account_id == 0 ? data.aws_caller_identity.id.account_id : var.account_id}"
-  assume_role_cmd = "source ${path.module}/assume_role.sh ${local.account_id} ${var.role}"
+  region          = "${var.region == 0 ? data.aws_region.current : var.region}"
+  assume_role_cmd = "source ${path.module}/assume_role.sh ${local.account_id} ${var.role} ${var.region}"
```

Comment:
if you were to prefix this command with AWS_DEFAULT_REGION=${var.region} then you would not need to make any modifications to the assume_role.sh shell script

Reply:
I've also just noticed that it is the create & destroy and commands that will need the region injected, not the assume role command. I.e the purpose of the region attribute on this module will be to create a given resource in the specified region.